fedimint-server/src/config/api.rs

```diff
@@ -481,6 +486,61 @@ impl ConfigGenApi {
             sleep(Duration::from_millis(100)).await;
         }
     }
+
+    // Check the status of the bitcoin rpc connection
+    pub async fn check_bitcoin_status(&self) -> ApiResult<BitcoinRpcConnectionStatus> {
+        let bitcoin_rpc_config = BitcoinRpcConfig::get_defaults_from_env_vars().map_err(|e| {
+            ApiError::server_error(format!("Failed to get bitcoin rpc env vars: {e}"))
+        })?;
+        match bitcoin_rpc_config.kind.as_str() {
+            "bitcoind" => check_bitcoind_status(&bitcoin_rpc_config),
+            "esplora" => self.check_esplora_status(&bitcoin_rpc_config).await,
+            _ => Err(ApiError::bad_request(format!(
+                "Unsupported bitcoin rpc kind: {}",
+                bitcoin_rpc_config.kind
+            ))),
+        }
+    }
+
+    async fn check_esplora_status(
+        &self,
+        bitcoin_rpc_config: &BitcoinRpcConfig,
+    ) -> ApiResult<BitcoinRpcConnectionStatus> {
+        let client =
+            create_bitcoind(bitcoin_rpc_config, self.task_group.make_handle()).map_err(|e| {
+                ApiError::server_error(format!("Failed to connect to bitcoin rpc: {e}"))
+            })?;
+        // TODO: Find a better way to check if esplora is synced, just returning Synced
+        // if it connects
+        let _network_info = client
+            .get_network()
+            .await
+            .map_err(|e| ApiError::server_error(format!("Failed to get esplora info: {e}")))?;
+        Ok(BitcoinRpcConnectionStatus::Synced)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Copy)]
+pub enum BitcoinRpcConnectionStatus {
+    Synced,
+    Syncing(f64),
+}
+
+fn check_bitcoind_status(
+    bitcoin_rpc_config: &BitcoinRpcConfig,
+) -> ApiResult<BitcoinRpcConnectionStatus> {
+    let (url, auth) = fedimint_bitcoind::bitcoincore::from_url_to_url_auth(&bitcoin_rpc_config.url)
+        .map_err(|e| ApiError::server_error(format!("Failed to parse bitcoin rpc url: {e}")))?;
+    let client = bitcoincore_rpc::Client::new(&url, auth)
+        .map_err(|e| ApiError::server_error(format!("Failed to connect to bitcoin rpc: {e}")))?;
+    let blockchain_info = block_in_place(|| client.get_blockchain_info())
+        .map_err(|e| ApiError::server_error(format!("Failed to get blockchain info: {e}")))?;
+    if blockchain_info.initial_block_download {
```

Comment:
What if the initial block download completed, but the node was turned off for an extended period of time and is now unsynced again?

Reply:
fixed, returns syncing if <1.0 for verification_progress, ibd vs normal syncing delay doesnt matter since we just return synced percentage